src/main/scala/io/iohk/ethereum/mining/BlockGenerator.scala

```diff
@@ -75,6 +73,22 @@ class BlockGenerator(blockchainStorages: BlockchainStorages, blockchainConfig: B
         }
       }.getOrElse(Left(NoParent))
     }
+
+    result.right.foreach(b => cache.updateAndGet(new UnaryOperator[List[Block]] {
+      override def apply(t: List[Block]): List[Block] =
+        (b :: t).take(cacheLimit)
+    }))
```

Comment:
Maybe the question shouldn't be "can we?`, but "should we?". It's up to us what we mine. We can simply refuse to give new work to `ethminer` until we received the mined block.